src/API/Server.hs

```diff
@@ -0,0 +1,122 @@
+{-# LANGUAGE DataKinds       #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeOperators   #-}
+-- {-# LANGUAGE OverloadedStrings #-}
+
+--
+-- Server.hs
+--
+-- Handles starting up an instance of Servant
+-- Returns encoded board data on request, and computes
+-- boards when input data is taken
+--
+module API.Server (startServer, serverApp) where
+
+import Control.Concurrent
+import Control.Monad
+import Data.Aeson
+import Data.Aeson.TH
+import Network.Wai
+import Network.Wai.Handler.Warp
+import Servant
+import Parser.Parser
+import Language.Syntax
+import Runtime.Repl
+import Runtime.Typechecker
+import Runtime.Eval
+
+-- for liftIO apparently????
+import qualified Graphics.UI.Threepenny as UI
+
+-- representation of input to the repl, from the user
+data SpielCommand = SpielCommand {
+    file   :: String,
+    inputs :: [String]
+  } deriving (Eq, Show)
+
+-- representation of the response from the Repl
+data SpielResponse = SpielResponse {
+    responses :: [String]
```

Comment:
Responses should have type [Val], or really [Either Exception Val] to handle the 'needs input' case